specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,175 @@
+
+TODO: edit the third bullet to just say that it needs to be able to produce one of the define output formats? Are all output formats equivalent (I think so, *if* the labeling scheme is defined)?  
+-> "the order requested by the program" may be somewhat ambiguous in any case, if we want to allow for parallelism/async execution
+
+The second requirement should be taken to mean that a Base Profile compilant program does *not* apply instructions to a qubit after it has been measured; all instructions to transform the quantum state can be applied before performing any measurements. It specifically also means that there is no need for the QPU to be able to measure only a subset of all available qubits at a time.
+
+## Program Structure
+
+A Base Profile compliant program is defined in the form of a single LLVM bitcode file that consists of the following:
+- definition of the opaque `Qubit` and `Result` types
+- [entry point definition](#entry-point-definition)
+- declarations of [QIS functions](#quantum-instruction-set)
+- declarations of functions used for [output recording](#output-recording)
+- [attributes](#attributes)
+
+The human readable LLVM IR for the bitcode can be obtained using standard 
+[LLVM tools](https://llvm.org/docs/CommandGuide/llvm-dis.html). For the purpose of clarity, this specification contains examples of the human readable IR emitted by [LLVM 13](https://releases.llvm.org/13.0.1/docs/LangRef.html). While the bitcode reprentation is portable and usually backward compatible, there may be visual differences in the human readable format depending on the LLVM version. This differences are irrelevant when using standard tools to load, manipulate, and/or execute bitcode.
+
+The code below illustrates how a simple program looks like within a Base Profile representation:
+
+```llvm
+; type definitions
+
+%Result = type opaque
+%Qubit = type opaque
+
+; global constants (labels for output recording)
+
+@0 = internal constant [16 x i8] c"label_schema_id\00"
+@1 = internal constant [3 x i8] c"r1\00"
+@2 = internal constant [3 x i8] c"r2\00"
+
+; entry point definition
+
+define i64 @Entry_Point_Name() #0 {
+entry:
+
+  ; calls to QIS functions  
+  tail call void @__quantum__qis__h__body(%Qubit* null)
+  tail call void @__quantum__qis__cnot__body(%Qubit* null, %Qubit* inttoptr (i64 1 to %Qubit*))
+  tail call void @__quantum__qis__mz__body(%Qubit* null, %Result* null)
+  tail call void @__quantum__qis__mz__body(%Qubit* inttoptr (i64 1 to %Qubit*), %Result* inttoptr (i64 1 to %Result*))
+
+  ; calls to record the program output
+  tail call void @__quantum__rt__initialize_record_output(i8* getelementptr inbounds ([16 x i8], [16 x i8]* @0, i32 0, i32 0))  
+  tail call void @__quantum__rt__tuple_record_output(i64 2, i8* null)
+  tail call void @__quantum__rt__result_record_output(%Result* null, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @1, i32 0, i32 0))
```

Comment:
Pointer type `i8*` is not allowed by any provider at this point.

Reply:
With this update, it would meant that string labels are always injected and allowed (only) as arguments in output recording functions. A backend may ignore them. See the doc for more detail.

Reply:
How come the choice for string labels as opposed to integer labels? Integer labels seems like a better compatibility choice assuming that these instructions might one day be executed directly on the QPU.